pkg/webhook/v1alpha3/pod/utils.go

```diff
@@ -74,3 +80,64 @@ func getLabel(pod *v1.Pod, targetLabel string) (string, error) {
 	}
 	return "", errors.New("Label " + targetLabel + " not found.")
 }
+
+func getRemoteImage(pod *v1.Pod, namespace string, containerIndex int) (crv1.Image, error) {
+	// verify the image name, then download the remote config file
+	c := pod.Spec.Containers[containerIndex]
+	ref, err := name.ParseReference(c.Image, name.WeakValidation)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse image %q: %v", c.Image, err)
+	}
+	imagePullSecrets := []string{}
+	for _, s := range pod.Spec.ImagePullSecrets {
+		imagePullSecrets = append(imagePullSecrets, s.Name)
+	}
+	kc, err := k8schain.NewInCluster(k8schain.Options{
+		Namespace:          namespace,
+		ServiceAccountName: pod.Spec.ServiceAccountName,
+		ImagePullSecrets:   imagePullSecrets,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create k8schain: %v", err)
+	}
+
+	mkc := authn.NewMultiKeychain(kc)
+	img, err := remote.Image(ref, remote.WithAuthFromKeychain(mkc))
```

Comment:
What will be created when we call it? Will we get the image from the remote registry in a blocking manner?

Reply:
it only fetches manifest instead of image file